src/Spago/Build.hs

```diff
@@ -82,11 +83,23 @@ build BuildOptions{..} maybePostBuild = do
     DoInstall -> Fetch.fetchPackages cacheConfig deps packagesMinPursVersion
     NoInstall -> pure ()
   let allGlobs = Packages.getGlobs deps depsOnly configSourcePaths <> sourcePaths
+      genCoreFnOpts = [ Purs.ExtraArg "--codegen", Purs.ExtraArg "corefn" ]
       buildAction = do
-        Purs.compile allGlobs pursArgs
-        case maybePostBuild of
-          Just action -> action
-          Nothing     -> pure ()
+        case alternateBackend of 
+          Nothing ->
+              Purs.compile allGlobs pursArgs
+          Just backend -> do
+              let normalizedArgs = concatMap (fmap Purs.ExtraArg . Text.words . Purs.unExtraArg) pursArgs
+              when (genCoreFnOpts `List.isInfixOf` normalizedArgs) $ 
```

Comment:
`purs` would still fail if the user passes e.g. `--codegen docs`, so let's check just for `--codegen` being in the array of `pursArgs`

Reply:
good call